### PR TITLE
ci: add a workflow that lints commit messages

### PR DESCRIPTION
**Pull request by sfan5**
works on PRs and normally pushed commits (in a branch)

here's an example how it looks https://github.com/sfan5/mpv/pull/1

to get a feeling for the rules, here's all commits since last release thrown into it: https://0x0.st/HXrp.txt
most are directly and objectively grounded in contribute.md, some other could be argued to be personal preference

### CODE REVIEW

**Comment by llyyr** [comment]
```
     DOCS/client-api-changes.rst: move mpv_del_property to 0.36.0
    
    incorrectly placed under 0.35.0 by 2cfaa820e54b12de437bb4da97d894ad610abcdd

has the following issues:
* Subject line must contain a prefix identifying the sub system [subsystem_prefix]
```

This is present in the text file but passes with the subsystem_prefix function in the Python script. How?

File: ci/lint-commit-msg.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+# been seeing this one all over github lately, must be the webshits
+@lint_rule("Do not use 'conventional commits' style")
+def no_cc(body):
+	return not re.search(r"(?i)^(feat|fix|chore|refactor)[!:(]", body[0])
+
+@lint_rule("History must be linear, no merge commits")
+def no_merge(body):
+	return not body[0].startswith("Merge ")
+
+@lint_rule("Subject line should be shorter than 90 characters")
```
Comment (christoph-heinrich) [inline ci/lint-commit-msg.py:96]: contribute.md says 72 characters

**Comment by sfan5** [comment]
@llyyr the subject line starts with a space

File: ci/lint-commit-msg.py (hunk 1 above)
Comment (sfan5) [reply ci/lint-commit-msg.py:96]: I went with 72 first but found it is more often broken when checking the results on existing commits.

Comment (christoph-heinrich) [reply ci/lint-commit-msg.py:96]: Just because past commits didn't strictly adhere to that doesn't mean we shouldn't check for it in future commits.

Comment (Dudemanguy) [reply ci/lint-commit-msg.py]: Keeping subject lines under 72 characters can get really impractical sometimes. I generally try to adhere to it myself, but there are times where it's better to type it out imo.

Comment (christoph-heinrich) [reply ci/lint-commit-msg.py]: It doesn't prevent merging afaik, so if there is a good reason it can still exceed that limit.

**Pull request merged**